lib/tcp.h

```diff
@@ -2,9 +2,14 @@
 
 #include <json-c/json.h>
 
+// Takes ownership of `msg`.
```

Comment:
For my edification: This is 'ownership' insofar as the sibling function (`msg_recv_listener`) ultimately frees it [here](https://github.com/khollbach/gossip-glomers-c/blob/d036b168066affb9396889e3fae787da2e64062c/lib/tcp.c#L117)?

Reply:
Exactly! Locally-speaking, ownership first goes into the function `msg_send_pusher` (specifically the local variable `msg` "owns" the data). And then inside that function, ownership transfers to the `Message` object as the `->msg` field. And then into the `Queue` when we call `enqueue`.

When we pop that particular element from the queue in `recv_listener`, ownership is passed into that function, which immediately frees it. (I'd even say that `free` takes ownership, just permanantly.)